Update DEVELOPEMENT.md to include section regarding gotestsum (#936)
DataDog/terraform-provider-datadog: DEVELOPMENT.md

-   [Go][3] 1.15 (to build the provider plugin)
-   A clone of this repository and the [\$GOPATH environment variable][7] set
-   [tfplugindocs][8]

> -   [gotestsum][9] (to run project tests) `gotestsum` executable binary is installed into `$GOPATH/bin` directory if the command is not available. For convenience, add the bin directory to your `PATH`

Reviewer: ```suggestion
-   [gotestsum][9] (to run project tests) `gotestsum` executable binary is installed into `$GOPATH/bin` when running `make get-test-deps`. Add the $GOPATH/bin directory to your `$PATH`
```

Reviewer: Its required that GOPATH/bin is in your path, lets use stronger wording 🙂
